XCUITests/TabCounterTests.swift

```diff
@@ -0,0 +1,52 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import XCTest
+
+class TabCounterTests: BaseTestCase {
+
+    func testTabIncrement() throws {
+        navigator.nowAt(NewTabScreen)
+        navigator.performAction(Action.CloseURLBarOpen)
+        waitForTabsButton()
+
+        var tabsOpen = app.buttons["Show Tabs"].value
+        XCTAssertEqual("1", tabsOpen as? String)
+
+        navigator.createNewTab()
+        navigator.nowAt(NewTabScreen)
+        navigator.performAction(Action.CloseURLBarOpen)
+        waitForTabsButton()
+
+        tabsOpen = app.buttons["Show Tabs"].value
+        XCTAssertEqual("2", tabsOpen as? String)
+    }
+
+    func testTabDecrement() throws {
+        navigator.nowAt(NewTabScreen)
+        navigator.performAction(Action.CloseURLBarOpen)
+        waitForTabsButton()
+
+        var tabsOpen = app.buttons["Show Tabs"].value
+        XCTAssertEqual("1", tabsOpen as? String)
+
+        navigator.createNewTab()
+        navigator.nowAt(NewTabScreen)
+        navigator.performAction(Action.CloseURLBarOpen)
+        waitForTabsButton()
+
+        tabsOpen = app.buttons["Show Tabs"].value
+        XCTAssertEqual("2", tabsOpen as? String)
+
+        navigator.goto(TabTray)
+        navigator.performAction(Action.AcceptRemovingAllTabs)
```

Comment:
can you please add a step before this one in which you only remove one tab (using the 'x' button) in the tab tray and check that the counter decrements? that's the issue we have seen several times, that removing tab by tab the counter is not correctly updated.
Thanks!

Reply:
Replaced the close all tabs with a close button action instead.